sdk/go/common/resource/plugin/converter_plugin.go

```diff
@@ -129,8 +129,17 @@ func (c *converter) ConvertState(ctx context.Context, req *ConvertStateRequest)
 		}
 	}
 
+	// Translate the rpc diagnostics into hcl.Diagnostics.
+	var diags hcl.Diagnostics
+	for _, rpcDiag := range resp.Diagnostics {
+		diags = append(diags, RPCDiagnosticToHclDiagnostic(rpcDiag))
```

Comment:
Just for my edification, what's the difference between RPC Diagnostics and HCL diagnostics?  It looks like at different points we're sending either type over RPCs, so I guess they are distinguished for some other reason?

Reply:
HCL diagnostics are a type specified by the library we use to handle HCL code parsing and we use it our codegen logic a lot. We also expose codegen facilities across the gRPC layer which may need communicate the diagnostics back and forth, so we translate the HCL type to a specific gRPC type (or vice-versa) that represents the same structure.